code/modules/recycling/disposal-unit.dm

```diff
@@ -527,6 +527,179 @@
 		M.forceMove(src)
 	flush()
 
+//Trap Door
+/obj/machinery/disposal/trapdoor
+	name = "trapdoor"
+	desc = "Almost like a door, but on floor."
+	density = 0
+	icon = 'icons/obj/atmospherics/pipes/trapdoor.dmi'
+	icon_state = "closed"
+	mode = 0
+	layer = 2.01
+	var/id = 1
+	var/auto_close = 1200
+	var/auto_close_on_mob = 100
+	var/sound_open = 'sound/machines/blast_door.ogg'
+	var/sound_close = 'sound/machines/blast_door.ogg'
+	var/open = FALSE
+
+/obj/machinery/disposal/trapdoor/proc/open()
+	if(flushing)
+		return
+	if(open)
+		return
+	open = TRUE
+	flick("opening", src)
+	playsound(src.loc, sound_open, 100, 1)
+	icon_state = "open"
+	spawn(5)
+		for(var/mob/living/M in loc)
+			if(!M.floating)
+				M.loc = src
+				trap_flush()
+				if(auto_close_on_mob)
+					spawn(auto_close_on_mob)
+						close()
+		for(var/obj/item/O in loc)
+			if(!O.throwing || !O.anchored)
+				O.loc = src
+					trap_flush()
+	if(auto_close)
+		spawn(auto_close)
+			close()
+	return 1
+
+
+/obj/machinery/disposal/trapdoor/proc/close()
+	if(open)
+		flick("closing", src)
+		icon_state = "closed"
+		playsound(src.loc, sound_open, 100, 1)
+		open = FALSE
+		return 1
+
+/obj/machinery/disposal/trapdoor/New(loc,var/obj/structure/disposalconstruct/make_from)
+	..()
+	stored.ptype = DISP_END_CHUTE
+	spawn(5)
+		trunk = locate() in loc
+		if(trunk)
+			trunk.linked = src
+
+/obj/machinery/disposal/trapdoor/Crossed(AM as mob|obj)
+	if(open)
+		if(istype(AM, /mob/living/))
+			var/mob/living/M = AM
+			if(M.floating)
+				return
+			M.loc = src
+			trap_flush()
+			if(auto_close_on_mob)
+				spawn(auto_close_on_mob)
+					close()
+			return
+		if(istype(AM, /obj/item))
+			var/obj/item/O = AM
+			spawn(5)
+				if(O.throwing || O.anchored)
+					return
+				else if(O.loc == src.loc)
+					O.loc = src
+					trap_flush()
+
+/obj/machinery/disposal/trapdoor/MouseDrop_T(mob/living/target, mob/living/user)
+	if (!open)
+		return
+	if(istype(target))
+		push_mob_in(target, user)
+		return 1
+
+/obj/machinery/disposal/trapdoor/proc/push_mob_in(mob/living/target, mob/living/carbon/human/user)
+	if(target.buckled)
+		return
+	add_fingerprint(user)
+	if(user == target)
+		if(target.floating)
+			user.visible_message("[user] is attempting to dive into [src].", \
+				"<span class='notice'>You start diving into [src]...</span>")
+			if(!do_mob(target, user, 10))
+				return
+			target.loc = src
+			user.visible_message("[user] dives into [src].", \
+				"<span class='notice'>You dive into [src].</span>")
+			sleep(5)
+			trap_flush()
+			if(auto_close_on_mob)
+				spawn(auto_close_on_mob)
+					close()
+		else
+			user.visible_message("[user] is attempting to step on the edge of [src].", \
+				"<span class='notice'>You start attempting to step on the edge of [src]...</span>")
+			if(!do_mob(target, user, 30))
+				return
+			var/chance = 25 // normal chance, 25% to fall inside
+			var/turf/open/floor/T = get_turf(src)
+			var/M = "fall inside"
+			var/U = "falls inside"
```

Comment:
instead of having 2 vars,you can use trinary operators like condition ? iftrue : iffalse like  user == target ? "fall" : "falls"